R/zchunk_L202.an_input.R

```diff
@@ -0,0 +1,646 @@
+    A_regions <- get_data(all_data, "energy/A_regions")
+    A_agRsrc <- get_data(all_data, "aglu/A_agRsrc")
+    A_agRsrcCurves <- get_data(all_data, "aglu/A_agRsrcCurves")
+    A_agUnlimitedRsrcCurves <- get_data(all_data, "aglu/A_agUnlimitedRsrcCurves")
+    A_an_input_supplysector <- get_data(all_data, "aglu/A_an_input_supplysector")
+    A_an_input_subsector <- get_data(all_data, "aglu/A_an_input_subsector")
+    A_an_input_technology <- get_data(all_data, "aglu/A_an_input_technology")
+    A_an_input_globaltech_shrwt <- get_data(all_data, "aglu/A_an_input_globaltech_shrwt")
+    A_an_supplysector <- get_data(all_data, "aglu/A_an_supplysector")
+    A_an_subsector <- get_data(all_data, "aglu/A_an_subsector")
+    A_an_technology <- get_data(all_data, "aglu/A_an_technology")
+    L132.ag_an_For_Prices <- get_data(all_data, "L132.ag_an_For_Prices")
+
+    # 2. Build tables
+    # Base table for resources - add region names to Level1 data tables (lines 49-70 old file)
+
+    # Following datasets are already 'long' so just skip the old interpolate_and_melt step
+    get_join_filter <- function(x) {   # internal function as we do these steps 5 times below
+      get_data(all_data, x) %>%
+        left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") %>%
+        filter(year %in% BASE_YEARS)
+    }
+    L202.an_Prod_Mt_R_C_Sys_Fd_Y.mlt <- get_join_filter("L107.an_Prod_Mt_R_C_Sys_Fd_Y")
+    L202.an_FeedIO_R_C_Sys_Fd_Y.mlt <- get_join_filter("L107.an_FeedIO_R_C_Sys_Fd_Y")
+    L202.an_Feed_Mt_R_C_Sys_Fd_Y.mlt <- get_join_filter("L107.an_Feed_Mt_R_C_Sys_Fd_Y")
+    L202.ag_Feed_Mt_R_C_Y.mlt <- get_join_filter("L108.ag_Feed_Mt_R_C_Y")
+    L202.an_ALL_Mt_R_C_Y <- get_join_filter("L109.an_ALL_Mt_R_C_Y")
+
+    # L202.RenewRsrc: generic resource attributes
+    A_agRsrc %>%
+      write_to_all_regions(LEVEL2_DATA_NAMES[["RenewRsrc"]], GCAM_region_names) %>%
+      mutate(market = if_else(market == "regional", region, market)) ->
+      L202.RenewRsrc
+
+    # L202.RenewRsrcPrice: resource prices
+    L202.RenewRsrc %>%
+      select(region, renewresource) %>%
+      mutate(year = min(BASE_YEARS), price = 1) ->
+      L202.RenewRsrcPrice
+
+    # L202.maxSubResource: maximum amount of resource production allowed in any period (72-97)
+    # Compute the maxsubresource as the maximum of all base periods, for each region and resource
+    # Note that the supply curves can exceed this number, by setting "available" to a number > 1.
+    # In this computation, we're using the sub.renewable.resource for name matching because the resource
+    # for scavenging is assigned a different name from the corresponding commodity and supplysector
+    # (to avoid having two markets with the same name)
+    L202.an_ALL_Mt_R_C_Y %>%
+      filter(GCAM_commodity %in% A_agRsrcCurves$sub.renewable.resource) %>%
+      group_by(region, GCAM_region_ID, GCAM_commodity) %>%
+      summarise(value = max(Prod_Mt)) ->
```

Comment:
I'm assuming this column name has to be `value`? Otherwise something more informative would be nice